activestorage/lib/active_storage/service/azure_storage_service.rb

```diff
@@ -123,6 +108,22 @@ def headers_for_direct_upload(key, content_type:, checksum:, filename: nil, disp
       { "Content-Type" => content_type, "Content-MD5" => checksum, "x-ms-blob-content-disposition" => content_disposition, "x-ms-blob-type" => "BlockBlob" }
     end
 
+    protected
+      def private_url(key, expires_in:, filename:, disposition:, content_type:, **)
+        signer.signed_uri(
+          uri_for(key), false,
+          service: "b",
+          permissions: "r",
+          expiry: format_expiry(expires_in),
+          content_disposition: content_disposition_with(type: disposition, filename: filename),
+          content_type: content_type
+        ).to_s
+      end
+
+      def public_url(key, **)
+        blobs.send(:blob_uri, container, key).to_s
```

Comment:
Do we need to `send` here, or is there a public API we can use?

Reply:
Thanks for catching this. I looked through the code in `azure-storage-blob` and I've realized that the method `uri_for` below in this file does the exact same thing.